src/sentry/tasks/statistical_detectors.py

```diff
@@ -92,14 +114,21 @@ def run_detection() -> None:
     performance_projects_count = 0
     profiling_projects_count = 0
 
-    for project in RangeQuerySetWrapper(
+    projects = RangeQuerySetWrapper(
         Project.objects.filter(status=ObjectStatus.ACTIVE).select_related("organization"),
         step=100,
-    ):
+    )
+
+    performance_project_settings = get_performance_project_settings(
+        [project.id for project in projects]
+    )
+
+    for project in projects:
```

Comment:
Because `RangeQuerySetWrapper` produces an iterator, this effectively runs 2 scans over the projects table. The suggestion I proposed yesterday would only run 1 scan. If you wish to return the settings with it instead of filtering in the same step, we could do something like

```python
def all_projects_with_settings():
    for projects in chunk(
        RangeQuerySetWrapper( 
            Project.objects.filter(status=ObjectStatus.ACTIVE).select_related("organization"),
            step=100,
        )
        100,
    ):
        settings = get settings for projects batch
        for project, setting in zip(projects, settings):
            yield project, setting
```

Then call it like
```python
for project, setting in all_projects_with_settings():
    ...
```

Reply:
oh I like this pattern 👍